dask/array/core.py

```diff
@@ -3493,6 +3498,72 @@ def _vindex(x, *indexes):
     return result_1d.reshape(broadcast_shape + result_1d.shape[1:])
 
 
+def _vindex_dask_arrays(x, *indices):
+    """Point wise indexing with Dask Arrays"""
+
+    from .creation import arange
+
+    x = asarray(x)
+
+    oth_inds = []
+    arr_inds = []
+    out_axes = []
+    for i, ind in enumerate(indices):
+        if isinstance(ind, Array):
+            oth_inds.append(slice(None))
+            arr_inds.append(asarray(ind))
+            if 0 not in out_axes:
+                out_axes.append(0)
+        elif isinstance(ind, Integral):
+            oth_inds.append(ind)
+        else:
+            oth_inds.append(ind)
+            arr_inds.append(slice(None))
+            out_axes.append(len(arr_inds))
+    oth_inds = tuple(oth_inds)
+    arr_inds = tuple(arr_inds)
+    out_axes = tuple(out_axes)
+
+    if not all(o == slice(None) for o in oth_inds):
+        x = x.vindex[oth_inds]
+
+    x_inds = tuple(
+        arange(s, chunks=c) for s, c in zip(x.shape, x.chunks)
+    )
+
+    idx_vals = atop(
+        _vindex_dask_arrays_chunk,
+        out_axes,
+        x, tuple(range(1, 1 + x.ndim)),
+        *concat([
+            (e_0, (0,), e_i, (i,)) if isinstance(e_0, Array) else
+            (e_0, None, e_i, (i,))
+            for i, (e_0, e_i) in enumerate(zip(arr_inds, x_inds), start=1)
+        ]),
+        dtype=x.dtype,
+        concatenate=True
+    )
+
+    return idx_vals
+
+
+def _vindex_dask_arrays_chunk(xc, *args):
```

Comment:
It would be helpful if this function had a docstring with example input/output.

Reply:
Added a docstring. Missed the example input/output part when writing the docs/comments. Sorry. Happy to add that as well in a follow-up.

In the interim it would be good to know if the docs/comments added help clarify this function and its arguments. Also would be good to hear how to better present that information.